Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabDomUtils.cpp

```diff
@@ -582,6 +582,113 @@ namespace AzToolsFramework
+                        result = false;
+                        AZ_Error("Prefab", false, "'%s' lacks 'Components' container in Entity (%s, '%s').",
+                            path.c_str(), entityAlias.c_str(), entityName.c_str());
+                        continue;
+                    }
+
+                    // Search current entity object for TransformComponent looping through components
+                    for (auto componentIt = componentsIt->value.MemberBegin(); componentIt != componentsIt->value.MemberEnd();
+                         ++componentIt)
+                    {
+                        const auto componentsTypeIt = componentIt->value.FindMember(PrefabDomUtils::TypeName);
+                        if (componentsTypeIt == componentIt->value.MemberEnd() || !componentsTypeIt->value.IsString())
+                        {
+                            result = false;
+                            AZStd::string componentName = componentIt->name.GetString();
+                            AZ_Error("Prefab", false, "'%s' lacks '$type' sub-object in a component object of Entity (%s, '%s').",
+                                path.c_str(), entityAlias.c_str(), entityName.c_str());
+                            continue;
+                        }
+                        
+                        AZStd::string componentAlias(componentsTypeIt->value.GetString());
+                        if (!componentAlias.contains("TransformComponent"))
+                        {
+                            continue;
+                        }
+                        
+                        constexpr const auto parentObjectName = "Parent Entity";
+                        auto parentIt = componentIt->value.FindMember(parentObjectName);
+                        if (parentIt == componentIt->value.MemberEnd() || !parentIt->value.IsString())
+                        {
+                            result = false;
+                            AZ_Error("Prefab", false, "'%s' lacks 'Parent Entity' object in TransformComponent of Entity (%s, '%s').",
+                                path.c_str(), entityAlias.c_str(), entityName.c_str());
+                            break; // one TransformComponent for an Entity -> invalid
+                        }
+
+                        if (AZStd::string(parentIt->value.GetString()).empty()) // parent link empty ?
+                        {
+                            // Re-link this entity under the ContainerEntity using its Alias
+                            parentIt->value.SetString(rapidjson::StringRef(containerEntityAlias.c_str()), templateDomRef.GetAllocator());
+
+                            result = false; // report that changes were needed and made 
+                            AZ_Error("Prefab", false,
+                                "'%s' lacks 'Parent Entity' value in TransformComponent of Entity (%s, '%s')"
+                                "\nEntity re-linked under the root container '%s'.",
```

Comment:
As per the previous comment, if this is a situation where we automatically fixed this and its not going to be a problem anymore, its probably better as a warning, or even a printf than an error, unless this will definitely break something.